### PR TITLE
[Merged by Bors] - feat(chaining): added transformations chaining page

### PR DESCRIPTION
**Pull request by galibey**

### CODE REVIEW

File: content/docs/concepts/transformations-chain.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+title: Transformations
+weight: 100
+---
+
+In Fluvio, the data transformation is done using [SmartModules](../../../smartmodules/) - user-defined functions compiled to WASM. A group of SmartModules forms a transformation chain - sequential invocations of each SmartModule according to the order defined upon creation. The output of the first invocation becomes the input of the second, the output of the second becomes the input of the third, and so on.
```
Comment (ajhunyady) [inline content/docs/concepts/transformations-chain.md:6]: compiled to WASM => compiled to WebAssembly (WASM)

File: content/docs/concepts/transformations-chain.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Both Fluvio Producer and Fluvio Consumer support **Transformation Chaining**. However, the actual code execution happens in different places. For Producer, the transformation
+happens before the data is sent to SPU, inside the Producer's process, hence consuming the resources of the client. For Consumer, the transformation takes place on SPU before the data is sent
```
Comment (ajhunyady) [inline content/docs/concepts/transformations-chain.md:9]: sent to SPU => sent to **the topic on** the SPU

Comment (ajhunyady) [reply content/docs/concepts/transformations-chain.md:9]: consuming the resources => utilizing the resources

(to remove the potential confusion with `consumer` in the next sentence.

File: content/docs/concepts/transformations-chain.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+to Consumer.
```
Comment (ajhunyady) [inline content/docs/concepts/transformations-chain.md:10]: to Consumer => to the Consumers

File: content/docs/concepts/transformations-chain.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+transforms:
+  - uses: infinyon/jolt@0.1.0
+    with:
+      spec:
+        - operation: default
+          spec:
+            source: "http"
+```
+we have one transformation, which is a SmartModule named `infinyon/jolt@0.1.0`. 
+The name must match a SmartModule that existed in the Cluster:
```
Comment (ajhunyady) [inline content/docs/concepts/transformations-chain.md:32]: that existed in the Cluster => previously downloaded to the Cluster

**Review by ajhunyady — COMMENTED** [review]
Nice job. A few nits below.

Can you also provide an example for SmartModules used at the CLI?

File: content/docs/concepts/transformations-chain.md (hunk 2 above)
Comment (galibey) [reply content/docs/concepts/transformations-chain.md]: done

File: content/docs/concepts/transformations-chain.md (hunk 3 above)
Comment (galibey) [reply content/docs/concepts/transformations-chain.md]: done

**Comment by galibey** [comment]
> Nice job. A few nits below.
> 
> Can you also provide an example for SmartModules used at the CLI?

The idea was that this page would describe the core concept of transformations. More details (including examples) will be on dedicated sections of corresponding components. In this page the list:

**Comment by galibey** [comment]
> Nice job. A few nits below.
> 
> Can you also provide an example for SmartModules used at the CLI?

The idea was that this page would describe the core concept of transformations. More details (including examples) will be on dedicated sections of corresponding components. On this page, the list:
```
**Transformation Chaining** is available on the following components:
 1. Fluvio Client
 2. Fluvio CLI
 3. SmartConnectors
```
will point to these sections once they are available.

**Review by ajhunyady — APPROVED** [review]
Nice work!

**Review by ajhunyady — APPROVED** [review]
done

**Pull request closed without merging**